spirv-opt: skips if_conversion when dontflatten is set

sindney opened the pull request:
Hi, we may mark some if statement as [flatten]/[branch] when cross compiling shader to bypass certain bugs.
I found that if_conversion pass ignores [branch] tag when performing optimizations.
I guess if_conversion should skip the optimization when an if statement is marked as [branch]?

s-perron commented:
Sorry, I rebased your change to get all of the tests to pass, but I forgot I wanted you to make a change.  I hope it does not mess you up too much.

sindney commented:
> Sorry, I rebased your change to get all of the tests to pass, but I forgot I wanted you to make a change. I hope it does not mess you up too much.

Sorry for my late reply. Sure, I've added the required test case,
